data-prepper-core/src/main/java/org/opensearch/dataprepper/peerforwarder/codec/PeerForwarderCodec.java

```diff
@@ -0,0 +1,7 @@
+package org.opensearch.dataprepper.peerforwarder.codec;
+
+public interface PeerForwarderCodec<T> {
+    byte[] serialize(T events) throws Exception;
+
+    T deserialize(byte[] bytes) throws Exception;
```

Comment:
This could also return a `Collection<Event>` along with my broader refactoring comments.